modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,477 @@
+			LEFT JOIN $wpdb->terms AS t ON tt.term_id = t.term_id
+		WHERE
+			post_status='publish' AND post_type IN ( {$post_types_in_string} ) AND post_date_gmt > (%s - INTERVAL 2 DAY)
+		GROUP BY p.ID
+		ORDER BY p.post_date_gmt DESC LIMIT %d", $cur_datetime, $limit );
+
+	header( 'Content-Type: application/xml' );
+	ob_start();
+	echo '<?xml version="1.0" encoding="UTF-8"?>' . "\n";
+	?>
+	<!-- generator="jetpack" -->
+	<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	        xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd"
+	        xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+	        xmlns:news="http://www.google.com/schemas/sitemap-news/0.9"
+	        xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
+		>
+		<?php
+		$posts = $wpdb->get_results( $query );
+		foreach ( $posts as $post ):
+
+			// Add in filter to allow skipping specific posts
+			if ( apply_filters( 'jetpack_sitemap_news_skip_post', false, $post ) ) {
+				continue;
+			}
+
+			$GLOBALS['post']                       = $post;
+			$url                                   = array ();
+			$url['loc']                            = get_permalink( $post->ID );
+			$news                                  = array ();
+			$news['news:publication']['news:name'] = get_bloginfo_rss( 'name' );
+			if ( function_exists( 'get_blog_lang_code' ) ) {
+				$news['news:publication']['news:language'] = get_blog_lang_code();
+			}
+			$news['news:publication_date'] = jetpack_w3cdate_from_mysql( $post->post_date_gmt );
+			$news['news:title']            = get_the_title_rss();
+			if ( $post->keywords ) {
+				$news['news:keywords'] = html_entity_decode( ent2ncr( $post->keywords ), ENT_HTML5 );
+			}
+			$url['news:news'] = $news;
+
+			// Add image to sitemap
+			if ( current_theme_supports( 'post-thumbnails' ) && has_post_thumbnail( $post->ID ) ) {
+				$post_thumbnail_id  = get_post_thumbnail_id( $post->ID );
+				$post_thumbnail_src = wp_get_attachment_image_src( $post_thumbnail_id );
+				if ( $post_thumbnail_src ) {
+					$url['image:image'] = array ( 'image:loc' => esc_url( $post_thumbnail_src[0] ) );
+				}
+			}
+
+			$url = apply_filters( 'jetpack_sitemap_news_sitemap_item', $url, $post );
+
+			if ( empty( $url ) ) {
+				continue;
+			}
+
+			jetpack_print_sitemap_item( $url );
+		endforeach;
+		?>
+	</urlset>
+	<?php
+	$xml = ob_get_contents();
+	ob_end_clean();
+	if ( ! empty( $xml ) ) {
+		set_transient( 'jetpack_news_sitemap', $xml, DAY_IN_SECONDS );
+		echo $xml;
+	}
+
+	die();
+}
+
+/**
+ * Absolute URL of the current blog's sitemap
+ *
+ * @return string sitemap URL
+ */
+function jetpack_sitemap_uri() {
+	global $current_blog;
+
+	$domain = $current_blog->primary_redirect ? $current_blog->primary_redirect : $current_blog->domain;
+
+	return apply_filters( 'sitemap_location', home_url( '/sitemap.xml' ) );
```

Comment:
All new `apply_filters` calls should have the proper documentation, and should be prefixed with `jetpack_`